site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/ParticipantFilterHandler.java

```diff
@@ -0,0 +1,263 @@
+        }
+        else if( selectedFilter.contains( SiteConstants.PARTICIPANT_FILTER_TYPE_SECTION ) )
+        {
+            return SiteConstants.PARTICIPANT_FILTER_TYPE_SECTION;
+        }
+        else if( selectedFilter.contains( SiteConstants.PARTICIPANT_FILTER_TYPE_ROLE ) )
+        {
+            return SiteConstants.PARTICIPANT_FILTER_TYPE_ROLE;
+        }
+        else
+        {
+            return SiteConstants.PARTICIPANT_FILTER_TYPE_ALL;
+        }
+    }
+
+    /**
+     * Utility method to add all groups as entries in the map.
+     *
+     * @param entries the map to add entries to
+     * @param site the currently active site
+     * @param groupTitlePostfix the postfix to use for groups in the dropdown
+     */
+    private static void addAllGroupEntries( LinkedHashMap<String, String> entries, Site site, String groupTitlePostfix )
+    {
+        if( site != null )
+        {
+            List<Group> groups = (List<Group>) site.getGroups();
+            if( groups != null )
+            {
+                // Sort the groups before adding to the map
+                Collections.sort( groups, SORT_BY_GROUP_TITLE );
+                for( Group group : groups )
+                {
+                    String prop = group.getProperties().getProperty( SiteConstants.GROUP_PROP_WSETUP_CREATED );
+                    if( StringUtils.isNotBlank( prop ) && Boolean.TRUE.toString().equals( prop ) )
+                    {
+                        entries.put( SiteConstants.PARTICIPANT_FILTER_TYPE_GROUP + group.getId(), group.getTitle() + " " + groupTitlePostfix );
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Utility method to add all roles as entries in the map.
+     *
+     * @param entries the map to add entries to
+     * @param site the currently active site
+     * @param roleTitlePostfix the postfix to use for roles in the dropdown
+     */
+    private static void addAllRoleEntries( LinkedHashMap<String, String> entries, Site site, String roleTitlePostfix )
+    {
+        if( site != null )
+        {
+            // Sort the roles before adding to the map
+            List<Role> roles = new ArrayList<>( site.getRoles() );
+            Collections.sort( roles, SORT_BY_ROLE_TITLE );
+            for( Role role : roles )
+            {
+                entries.put( SiteConstants.PARTICIPANT_FILTER_TYPE_ROLE + role.getId(), role.getId() + " " + roleTitlePostfix );
+            }
+        }
+    }
+
+    /**
+     * Utility method to add all sections as entries in the map.
+     *
+     * @param entries the map to add entries to
+     * @param site the currently active site
+     */
+    private static void addAllSectionEntries( LinkedHashMap<String, String> entries, Site site )
+    {
+        if( site != null )
+        {
+            List<String> sectionIDs = SiteParticipantHelper.getProviderCourseList( site.getId() );
+            if( sectionIDs != null )
```

Comment:
if it possible for siteparticipanthelper to return null here? If so, can we make it return empty list instead?

Reply:
Not possible for it to return `null`, so I'll change it to a `!sectionIDs.isEmpty()`.